libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -383,6 +383,13 @@ protected virtual async Task<bool> ProcessEventAsync(ActionContext actionContext
                         activity = ObjectPath.GetPathValue<Activity>(dialogEvent, "Value");
                         break;
                     }
+
+                case AdaptiveEvents.SetTestOptions:
+                    {
+                        // We received an SetTestOptions event, set value into conversation.testOptions
+                        actionContext.State.SetValue(TestOptions.Kind, dialogEvent.Value);
+                        break;
+                    }
```

Comment:
Is this not possible to do with middle ware?  Ideally, we would not change anything in Microsoft.Bot.Builder.Dialogs.Adaptive and the middleware would be part of Adaptive.Testing.  Events support custom payloads an this would be custom.

Reply:
Don't do this here...do it in middleware

---
In reply to: [469416323](https://github.com/microsoft/botbuilder-dotnet/pull/4429#discussion_r469416323) [](ancestors = 469416323)